### PR TITLE
Refactor constraint application of MGTwoLevelTransfer

### PR DESCRIPTION
**Pull request by peterrum**
... so that it uses the new class `internal::MatrixFreeFunctions::ConstraintInfo`.

~depends on #13518 and #13519~

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 1]
```diff
@@ -3233,30 +2884,30 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
...
-          for (unsigned int v = 0; v < n_lanes_filled; ++v)
-            {
-              for (unsigned int i = 0; i < scheme.n_dofs_per_cell_coarse; ++i)
-                vec_coarse_ptr->local_element(indices_coarse_plain[i]) =
-                  evaluation_data_coarse[i][v];
-              indices_coarse_plain += scheme.n_dofs_per_cell_coarse;
-            }
+          set_dof_values_plain(cell_global,
+                               n_lanes_filled,
+                               scheme.n_dofs_per_cell_coarse);
```
[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:2909]: I can inline the three lambdas, but the motivation was to make clear which `FEEvaluation` functions these code snippets correspond.

[the remark this answers is not in the document]
Comment (kronbichler) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:2909]: I prefer the straight-line code. We document this reasonably well via the `VectorDistributorLocalToGlobal` and similar structs that get passed to `read_write_operation` in my opinion.

[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h]: I have inlined the functions.

**Comment by peterrum** [comment]
I have rebased this PR!

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 2]
```diff
@@ -407,22 +407,10 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
   mutable LinearAlgebra::distributed::Vector<Number> vec_coarse;
 
   /**
-   * Constraint-entry indices for performing manual
-   * constraint_coarse.distribute_local_to_global().
+   * TODO
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:410]: Can you write a short documentation here?

**Review by kronbichler — APPROVED** [review]
This is so much nicer now. :+1:

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.h]: Done!

**Pull request merged**